lib/core/text/abstract_text.nit

```diff
@@ -1489,16 +1485,32 @@ abstract class Buffer
 	#     b = new FlatBuffer.from("ab_c -ab0c ab\nc")
 	#     b.capitalize
 	#     assert b == "Ab_C -Ab0C Ab\nC"
-	fun capitalize do
+	#
+	#     b = new FlatBuffer.from("12345")
+	#     b.capitalize(source="foo")
+	#     assert b == "Foo45"
+	#
+	#     b = new FlatBuffer.from("preserve my ACRONYMS")
+	#     b.capitalize(keep_upper=true)
+	#     assert b == "Preserve My ACRONYMS"
+	fun capitalize(source: nullable Text, keep_upper: nullable Bool) do
```

Comment:
I'm not fan of such a signature.

First the approach seems very weird to me (I don't say I think it's bad).
Did you see a parameter that fallbacks to self elsewhere? I would like to know where you get such an inspiration.

Second, no other Text/String/Buffer method have this kind of approach.
This add more complexity (unPOLAness) to an already quite complex class hierarchy.

What is the opinion of other people on this?